Added tests to the project

krufab opened the pull request:
Added tests to the project, as per issue #100

slide commented:
The problem is that you aren't a trusted contributor, so the CI will not use your Jenkinsfile changes.

oleg-nenashev commented on tests/tests.bats:
```diff
@@ -0,0 +1,122 @@
+@test "[${FLAVOR}] use build args correctly" {
+  cd "${BATS_TEST_DIRNAME}"/.. || false
+
+  local TEST_VERSION="3.36"
```
Would be nice to avoid putting it here directly. It will become immediately obsolete otherwise (now we are on Remoting 3.40)

krufab replied:
Ok, but that is a test: I'd like to be sure that the image allows the specification of a some different version than the one hardcoded in the Dockerfile ARG VERSION=3.40

oleg-nenashev commented on tests/tests.bats:
```diff
@@ -0,0 +1,122 @@
+
+DOCKERFILE=Dockerfile
+JDK=8
+SLAVE_IMAGE=jenkins-slave
```
Please use "AGENT" instead of "SLAVE" for new code

krufab replied:
Ok, done

krufab commented:
@oleg-nenashev, there is an integration test in jnlp-slave.
I agree integration tests would be useful as well. However they might come later on...

@slide , i'll revert the build.sh so that the tests can be executed

slide commented:
@krufab I was not saying that you should revert build.sh, I am just telling you why the CI is failing. What I can do is submit a shadow PR with your same changes so that the CI will actually run using the Jenkinsfile changes, would that be ok with you?

krufab commented:
@slide, I guess it's ok, yes

slide commented:
@krufab See #102

krufab commented:
@slide, Bats doesn't seem to be installed on the test server. I reverted the build.sh file, removed make tests and now it builds. You can run the tests on your local machine and check if they are ok. When the build server will be able to run the tests, I'll re-enable them.

slide commented:
If you look at the master docker repo, you can see that bats is a submodule. That would be the way to go here too.

slide commented on build.sh:
```diff
@@ -1,20 +1,3 @@
-done
-
-echo "Build finished successfully"
+make build
```
This should also `make tests`

krufab replied:
Done.

slide commented:
I pulled in all the latest changes into my duplicate PR so we can see the Jenkinsfile changes in action

slide commented:
#102 is passing for all the checks including the tests

krufab commented:
Is it ok this PR or is there anything to change?

slide commented:
I am happy with this, if @oleg-nenashev has no objections, I think it can be merged. I will look at adding some simple pester based tests for the Windows side.
